### PR TITLE
v4 Chaos Overhaul

### LINKED ISSUE #1092: Convert Khorne to v4

### LINKED ISSUE #1105: Convert Beasts of Chaos to v4

### LINKED ISSUE #1107: Convert Tzeentch to v4

### LINKED ISSUE #1109: Convert Tamurkhans Horde  to v4

### LINKED ISSUE #1114: Convert Slaves To Darkness to v4

### LINKED ISSUE #1117: Convert Nurgle to v4

### PR DESCRIPTION
**Pull request by JohnSchramm**
Added v4 Beasts of Chaos (Fixes #1105, Fixes #1117, Fixes #1114, Fixes #1109, Fixes #1107, Fixes #1092)
Added v4 Legion of Azgorh
Added v4 Legion of Chaos Ascendant subfaction (Requires Chaos GA support to fully link)
Modified Chaos armies with imports.
Removed old army entries and functions.

### CODE REVIEW

File: src/factions/legion_of_azgorh/units.ts [hunk 1]
```diff
@@ -138,44 +131,24 @@ export const Units: TEntry[] = [
...
-      {
-        name: `Martial Contempt`,
-        desc: `You can use this command ability in your hero phase. If you do so, pick 1 enemy unit within 12" of a friendly model with this command ability. Until the start of your next hero phase, add 1 to wound rolls for attacks made by friendly Legion of Azgorh units that target that unit. The same enemy unit cannot be picked as the target of this command ability more than once per hero phase.`,
-        when: [HERO_PHASE],
-        command_ability: true,
-      },
-    ],
+  'Infernal Guard Castellan': {
+    mandatory: {
+      units: [keyPicker(CommandAbilities, ['Martial Contempt'])],
```
Comment (exonian) [inline src/factions/legion_of_azgorh/units.ts:147]:
```suggestion
      command_abilities: [keyPicker(CommandAbilities, ['Martial Contempt'])],
```

File: src/factions/legion_of_azgorh/units.ts [hunk 2]
```diff
@@ -138,44 +131,24 @@ export const Units: TEntry[] = [
...
-      {
-        name: `Favour of the Burning God`,
-        desc: `You can use this command ability at the start of the combat phase. If you do so, pick 1 friendly model with this command ability. Add 1 to hit rolls for attacks made with melee weapons by friendly Ba'hal units while they are wholly within 12" of that model until the end of that phase.`,
-        when: [START_OF_COMBAT_PHASE],
-        command_ability: true,
-      },
-    ],
+  'Bull Centaur Taur`Ruk': {
+    mandatory: {
+      units: [keyPicker(CommandAbilities, ['Favour of the Burning God'])],
```
Comment (exonian) [inline src/factions/legion_of_azgorh/units.ts:138]:
```suggestion
      command_abilities: [keyPicker(CommandAbilities, ['Favour of the Burning God'])],
```

File: src/factions/legion_of_azgorh/units.ts [hunk 3]
```diff
@@ -111,21 +108,17 @@ export const Units: TEntry[] = [
...
         desc: `Do not use the attack sequence for an attack made with the Blood of Hashut. Instead roll a D6. On a 2+ the target unit suffers D3 mortal wounds. If the target unit is a War Machine it suffers D6 mortal wounds instead of D3.`,
         when: [SHOOTING_PHASE],
       },
     ],
   },
-  {
-    name: `Shar'Tor the Executioner`,
+  'Shar`Tor the Executioner': {
+    mandatory: {
+      units: [keyPicker(CommandAbilities, ['Lord of the Ba`hal'])],
```
Comment (exonian) [inline src/factions/legion_of_azgorh/units.ts:120]:
```suggestion
      command_abilities: [keyPicker(CommandAbilities, ['Lord of the Ba`hal'])],
```

File: src/factions/legion_of_azgorh/units.ts [hunk 4]
```diff
@@ -90,16 +91,12 @@ export const Units: TEntry[] = [
...
-        when: [START_OF_BATTLESHOCK_PHASE],
-        command_ability: true,
-      },
     ],
   },
-  {
-    name: `Daemonsmith`,
+  Daemonsmith: {
+    mandatory: {
+      units: [keyPicker(Spells, ['Ash Storm'])],
```
Comment (exonian) [inline src/factions/legion_of_azgorh/units.ts:98]:
```suggestion
      spells: [keyPicker(Spells, ['Ash Storm'])],
```

File: src/factions/legion_of_azgorh/units.ts [hunk 5]
```diff
@@ -66,9 +65,11 @@ const SiegeArtilleryEffects = [
   },
 ]
 
-export const Units: TEntry[] = [
-  {
-    name: `Drazhoath the Ashen`,
+const Units = {
+  'Drazhoath the Ashen': {
+    mandatory: {
+      units: [keyPicker(CommandAbilities, ['Lord of the Black Fortress'])],
```
Comment (exonian) [inline src/factions/legion_of_azgorh/units.ts:71]:
```suggestion
      command_abilities: [keyPicker(CommandAbilities, ['Lord of the Black Fortress'])],
```

**Review by exonian — CHANGES_REQUESTED** [review]

**Comment by daviseford** [comment]
Merging this in quickly so we can keep iterating. If you had more changes, go ahead and make a new PR. Awesome work!

**Pull request merged**